src/Featurizers/CountVectorizerFeaturizer.cpp

```diff
@@ -15,136 +14,45 @@ namespace Featurizers {
-    // Version
-    Traits<std::uint16_t>::serialize(ar, 1); // Major
-    Traits<std::uint16_t>::serialize(ar, 0); // Minor
-
-    // Data
-    Traits<decltype(_labels)>::serialize(ar, _labels);
-    Traits<decltype(_binary)>::serialize(ar, _binary);
-    Traits<decltype(_lower)>::serialize(ar, _lower);
-    Traits<std::underlying_type<AnalyzerMethod>::type>::serialize(ar, static_cast<std::underlying_type<AnalyzerMethod>::type>(_analyzer));
-    Traits<decltype(_regex)>::serialize(ar, _regex);
-    Traits<decltype(_ngram_min)>::serialize(ar, _ngram_min);
-    Traits<decltype(_ngram_max)>::serialize(ar, _ngram_max);
+bool CountVectorizerTransformer::operator==(CountVectorizerTransformer const &other) const {
+    return dynamic_cast<TfidfVectorizerTransformer &>(*_pTransformer) == dynamic_cast<TfidfVectorizerTransformer &>(*other._pTransformer);
 }
 
-bool CountVectorizerTransformer::operator==(CountVectorizerTransformer const &other) const {
-    return _labels == other._labels
-        && _binary == other._binary
-        && _lower == other._lower
-        && _analyzer == other._analyzer
-        && _regex == other._regex
-        && _ngram_min == other._ngram_min
-        && _ngram_max == other._ngram_max;
+void CountVectorizerTransformer::save(Archive &ar) const {
+    _pTransformer->save(ar);
 }
 
 // ----------------------------------------------------------------------
 // ----------------------------------------------------------------------
 // ----------------------------------------------------------------------
 void CountVectorizerTransformer::execute_impl(typename BaseType::InputType const &input, typename BaseType::CallbackFunction const &callback) /*override*/ {
-    AppearanceMapType appearanceMap;
-
-    std::string processedInput(Components::DocumentDecorator(input, _lower, _analyzer, _regex, _ngram_min, _ngram_max));
-
-    _parse_func(
-        processedInput,
-        [&appearanceMap] (std::string::const_iterator iter_start, std::string::const_iterator iter_end) {
-
-            AppearanceMapType::iterator iter_appearance(appearanceMap.find(std::make_tuple(iter_start, iter_end)));
-
-            if (iter_appearance != appearanceMap.end()) {
-                ++iter_appearance->second;
-            } else {
-                appearanceMap.insert(std::make_pair(std::make_tuple(iter_start, iter_end), 1));
-            }
-        }
-    );
-
-    std::vector<SparseVectorEncoding<std::uint32_t>::ValueEncoding> result;
 
-    for (auto const & wordAppearance : appearanceMap) {
-        std::string const word = std::string(std::get<0>(wordAppearance.first), std::get<1>(wordAppearance.first));
+    _pTransformer->execute(
+        input,
+        [&callback](SparseVectorEncoding<std::float_t> obj) {
 
-        typename IndexMapType::const_iterator const      iter_label(_labels.find(word));
+            std::vector<SparseVectorEncoding<std::uint32_t>::ValueEncoding> values;

```

Comment:
>values [](start = 76, length = 6)

Pre-reserve the size of values since you know how many elements there will be once the vector is populated.